tracing-tower/Cargo.toml

```diff
@@ -4,8 +4,29 @@ version = "0.1.0"
 authors = ["Eliza Weisman <eliza@buoyant.io>"]
 edition = "2018"
 
+[features]
+default = ["tower-layer", "tower-util", "http"]
+
 [dependencies]
 tracing = "0.1"
 tracing-futures = { path = "../tracing-futures" }
 futures = "0.1"
 tower-service = "0.2"
+tower-layer = { version = "0.1", optional = true }
+tower-util = { version = "0.1", optional = true }
+http = { version = "0.1", optional = true }
```

Comment:
It seems strange to me to have `http` be one of the dependencies here. Why isn't it a `dev-dependency`? Is it important that we provide `tracing-tower-http`?

Reply:
My rationale process was that the `tracing-tower-http` crate could eventually be removed in favour of a feature-flag on this crate, to avoid having to reimplement as much code for e.g. the instrument extension traits.

Reply:
I guess I'm asking a higher-level question of "why do we even provide `tracing-tower-http`?"

Reply:
The rationale behind having that crate was basically just that HTTP is a pretty common use-case for `tower`, and I thought it made sense to have batteries-included support for tower services that handle HTTP requests. I don't think it's strictly _necessary_, but having ergonomic helper methods would be nice.

Perhaps this (eventually) ought to be something provided by the `tower-http` crate, rather than by `tracing`? At the time the `tracing-tower-http` crate was originally written, we were early enough in the prototyping process that none of this was really ready to integrate directly into `tower`.